### PR TITLE
[C++ Worker] Replace `Ray::xxx` with `ray::xxx` and update namespaces

### LINKED ISSUE #14469: [C++ worker] Move non-API code out of `ray::api` namespace
### Describe your feature request

Lots of non-API code is currently defined under `ray::api` namespace. We should only put user-accessible code in `ray::api`.

I'll just give some examples here:

* ActorCreator
* ActorTaskCaller
* Arguments
* RayRuntime
* Serializer
* TaskCaller

### LINKED ISSUE #17114: [C++ API] use namespace instead of Ray class
Ray::Task -> ::ray::Task
Ray::Actor -> ::ray::Actor

### PR DESCRIPTION
**Pull request by qicosmos**
## Why are these changes needed?
Provide more friendly API of c++ worker, move the API into ray namespace.
ray:: => CPP public
ray::internal => CPP private
ray::core => Core

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests

### CODE REVIEW

File: cpp/example/example.cc [hunk 1]
```diff
@@ -5,7 +5,7 @@
 #include <ray/api.h>
 
 /// using namespace
-using namespace ::ray::api;
+using namespace ray;
```
Comment (kfstorm) [inline cpp/example/example.cc:8]: Remove this line? Same as other places in C++ worker tests.

File: src/ray/rpc/grpc_client.h [hunk 2]
```diff
@@ -49,8 +49,8 @@ class GrpcClient {
     // Disable http proxy since it disrupts local connections. TODO(ekl) we should make
     // this configurable, or selectively set it for known local connections only.
     argument.SetInt(GRPC_ARG_ENABLE_HTTP_PROXY, 0);
-    argument.SetMaxSendMessageSize(RayConfig::instance().max_grpc_message_size());
-    argument.SetMaxReceiveMessageSize(RayConfig::instance().max_grpc_message_size());
+    argument.SetMaxSendMessageSize(::RayConfig::instance().max_grpc_message_size());
```
Comment (kfstorm) [inline src/ray/rpc/grpc_client.h:52]: I noticed before that `RayConfig` is in the global namespace. Maybe we can move it to `ray::core` in this PR.

File: src/ray/core_worker/actor_handle.h [hunk 3]
```diff
@@ -24,7 +24,7 @@
 #include "src/ray/protobuf/gcs.pb.h"
 
 namespace ray {
-
+namespace core {
```
Comment (kfstorm) [inline src/ray/core_worker/actor_handle.h:27]: This PR looks great overall. The only concern I have is that whether we should rename `ray` to `ray::core` as a whole.

Comment (scv119) [reply src/ray/core_worker/actor_handle.h:27]: +1

**Comment by scv119** [comment]
cc @ericl  this is a massive change; might worth take a look. For context: https://ray-distributed.slack.com/archives/G015KN0NBEJ/p1627441955005200

**Comment by ericl** [comment]
I'm OK with renaming ray to ray::core.

File: src/ray/core_worker/actor_manager.cc [hunk 4]
```diff
@@ -70,7 +70,7 @@ bool ActorManager::AddNewActorHandle(std::unique_ptr<ActorHandle> actor_handle,
                         caller_address, actor_id, actor_creation_return_id);
 }
 
-bool ActorManager::AddActorHandle(std::unique_ptr<ActorHandle> actor_handle,
+bool ActorManager::AddActorHandle(std::unique_ptr<core::ActorHandle> actor_handle,
```
Comment (kfstorm) [inline src/ray/core_worker/actor_manager.cc:73]: I think adding `using namespace ray::core;` in .cc files is much simpler. We may move more classes from `ray` to `ray::core` later.

**Review by rkooo567 — APPROVED** [review]
LGTM if tests pass

File: src/ray/common/ray_config.cc [hunk 5]
```diff
@@ -49,16 +52,19 @@ void RayConfig::initialize(const std::string &config_list) {
     }
     if (RAY_LOG_ENABLED(DEBUG)) {
       std::ostringstream oss;
-      oss << "RayConfig is initialized with: ";
+      oss << "ray::core::RayConfig is initialized with: ";
```
Comment (raulchen) [inline src/ray/common/ray_config.cc:55]: this is not needed.

File: src/ray/core_worker/actor_manager.cc [hunk 6]
```diff
@@ -16,6 +16,8 @@
 
 #include "ray/gcs/pb_util.h"
 
+using namespace ray::core;
```
Comment (raulchen) [inline src/ray/core_worker/actor_manager.cc:19]: shouldn't this file itself have `ray::core` namespace?

**Review by raulchen — CHANGES_REQUESTED** [review]
All core code should be put under `ray::core` namespace. 
For the files in Ray core, this PR should only change their namespaces or add `using ...`. Don't need to add `ray::core` everywhere.

File: cpp/src/ray/runtime/object/local_mode_object_store.cc [hunk 7]
```diff
@@ -43,7 +42,7 @@ void LocalModeObjectStore::PutRaw(std::shared_ptr<msgpack::sbuffer> data,
   auto status = memory_store_->Put(
       ::ray::RayObject(buffer, nullptr, std::vector<ObjectID>()), object_id);
   if (!status) {
-    throw RayException("Put object error");
+    throw ray::internal::RayException("Put object error");
```
Comment (raulchen) [inline cpp/src/ray/runtime/object/local_mode_object_store.cc:45]: `ray::internal::` can be removed

File: cpp/src/ray/runtime/logging.cc [hunk 8]
```diff
@@ -16,25 +16,23 @@
 #include <ray/util/logging.h>
 
 namespace ray {
-namespace api {
 
-class RayLogImpl : public RayLog, public ray::RayLog {
+class RayLogImpl : public RayLogger, public ray::RayLog {
```
Comment (raulchen) [inline cpp/src/ray/runtime/logging.cc:20]: Rename to RayLoggerImpl

File: cpp/include/ray/api/logging.h [hunk 9]
```diff
@@ -58,9 +74,8 @@ class RayLog {
   virtual std::ostream &Stream() = 0;
 };
 
-std::unique_ptr<RayLog> CreateRayLog(const char *file_name, int line_number,
-                                     RayLogLevel severity);
-bool IsLevelEnabled(RayLogLevel log_level);
+std::unique_ptr<RayLogger> CreateRayLog(const char *file_name, int line_number,
```
Comment (raulchen) [inline cpp/include/ray/api/logging.h:77]: CreateRayLogger

**Review by raulchen — APPROVED** [review]

**Pull request merged**